### ISSUE TITLE
NullPointer Error in method "EnvelopeSummary results = envelopesApi.createEnvelope(accountId, envelope)"

### ISSUE THREAD

**Issue by SunStriderxxx** [docusign/code-examples-java#4]
NullPointer Error in method "EnvelopeSummary results = envelopesApi.createEnvelope(accountId, envelope)", i user java-client 3.6.0, but when i send envelop, i got an NullPointer Exception, i fount in 3.8.0 there is an try catch with
"updateAccessToken()"
![image](https://user-images.githubusercontent.com/36763683/96867780-bab96580-149f-11eb-9900-fd6b3bb48da0.png)

, which not found in 3.6.0,
![image](https://user-images.githubusercontent.com/36763683/96867822-c60c9100-149f-11eb-8f5a-024dc5fda262.png)

how can i solve it?

**Comment by AaronWDS** [comment]
Hello there, we did indeed update the way this mechanism worked between versions 3.6.0 and 3.8.0. If possible I would update to 3.8.0.  As a workaround you can [apply this patch to set ](https://github.com/docusign/code-examples-java/commit/99acb63145b4e0b6ec0d6a4b11ea36a0013cd8ca) to your API Client object:

```
ApiClient apiClient = new ApiClient(basePath);
        apiClient.addDefaultHeader(HttpHeaders.AUTHORIZATION, BEARER_AUTHENTICATION + userAccessToken);
       // it is a workaround to NPE that occurs on versions before 3.7.0-BETA
        apiClient.addAuthorization("docusignAccessCode", null);
        EnvelopeDefinition envelope = makeEnvelope(signerEmail, signerName);
        EnvelopesApi envelopesApi = new EnvelopesApi(apiClient);
        EnvelopeSummary envelopeSummary = envelopesApi.createEnvelope(accountId, envelope);
        String envelopeId = envelopeSummary.getEnvelopeId();


 private static EnvelopeDefinition makeEnvelope(String signerEmail, String signerName) throws IOException {
        // Create a signer recipient to sign the document, identified by name and email
        // We set the clientUserId to enable embedded signing for the recipient
        Signer signer = new Signer();
        signer.setEmail(signerEmail);
        signer.setName(signerName);
        signer.clientUserId(SIGNER_CLIENT_ID);
        signer.recipientId("1");
        signer.setTabs(EnvelopeHelpers.createSingleSignerTab("/sn1/", ANCHOR_OFFSET_Y, ANCHOR_OFFSET_X));

        // Add the recipient to the envelope object
        Recipients recipients = new Recipients();
        recipients.setSigners(Arrays.asList(signer));

        EnvelopeDefinition envelopeDefinition = new EnvelopeDefinition();
        envelopeDefinition.setEmailSubject("Please sign this document");
        envelopeDefinition.setRecipients(recipients);
        Document doc = EnvelopeHelpers.createDocumentFromFile(DOCUMENT_FILE_NAME, DOCUMENT_NAME, "3");
        envelopeDefinition.setDocuments(Arrays.asList(doc));
        // Request that the envelope be sent by setting |status| to "sent".
        // To request that the envelope be created as a draft, set to "created"
        envelopeDefinition.setStatus(EnvelopeHelpers.ENVELOPE_STATUS_SENT);

        return envelopeDefinition;
    }
```

**Comment by SunStriderxxx** [comment]
i use jdk1.8 so i can't use client-3.8.0, i found that if i call apiClietn.setAccessToken will solve this, when the token expired, i will catch the exception and set it again. This method can solve this problem.
![image](https://user-images.githubusercontent.com/36763683/97103957-d6647c00-16ea-11eb-9da6-6aff0e6a2aa6.png)

Is there a better way can solve this?

**Comment by AaronWDS** [comment]
Not that I am aware of but the patch I've provided works with 3.6.0 and doesnt rely on the `setAccessToken method`.

**Comment by LarryKlugerDS** [comment]
My understanding of this issue is that it is fixed in v3.8.0.
Closing this issue. Please re-open this is still an issue.
